### ISSUE TITLE
Is byte array not allowed as a return type?

### ISSUE THREAD

**Issue by victoriorinescu** [protobuf-net/protobuf-net.Grpc#116]
`
        public async ValueTask<IEnumerable<byte>> GetFile()
        {
            return System.IO.File.ReadAllBytes("file");
        }
`

_
		'Status(StatusCode=Internal, Detail="Error starting gRPC call. ProtoException: Invalid wire-type; this usually means you have over-written a file without truncating or setting the length; see https://stackoverflow.com/q/2152978/23354")'

_

**Comment by mgravell** [comment]
right now, no; I need to improve the exception handling here, so that it becomes more obvious. Basically, you want to return something like a `File` where `File` *has* a `byte[]` - something like:

``` c#
[ProtoContract]
public class File {
    [ProtoMember(1)]
    public byte[] Payload {get;set;}
}
```

or similar; for large files, you might want to use file fragments; this can be done by changing from `File` to `IAsyncEnumerable<File>` (and sending multiple fragments, obviously)

**Comment by SeppPenner** [comment]
That was a question I wanted to ask as well, but now experienced already. It would be good to extend the docs at some point that variables and return types of gRPC methods **must** be of a type that uses `[ProtoContract]` or `[DataContract]` and not of a standard data type like `int`, `double` or something.

**Comment by mgravell** [comment]
If you install protobuf-net.BuildTools, it should spot most of these problems for you at build time. If it doesn't: we can fix that.

**Comment by SeppPenner** [comment]
Yes, `protobuf-net.BuildTools` might be a good solution as my data models are quite complex in some cases...
